src/main/java/cz/muni/fi/Component.java

```diff
@@ -0,0 +1,62 @@
+
+package cz.muni.fi;
+
+import jdk.nashorn.internal.objects.annotations.Getter;
+import jdk.nashorn.internal.objects.annotations.Setter;
+
+import javax.persistence.Enumerated;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+/**
+ * Author: Richard Hrmo
+ */
+public class Component {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Getter @Setter private Long id;
+
+    @Getter private String name;
+
+    @Getter private boolean isAvailable = false;
+
+    @Enumerated
+    @Getter private ComponentType componentType;
+
+    public Component(){}
+
+    public Component(String name, boolean isAvailable, ComponentType componentType){
+        this.name = name;
+        this.isAvailable = isAvailable;
+        this.componentType = componentType;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public boolean isAvailable() {
+        return isAvailable;
+    }
+
+    public void setAvailable(boolean available) {
+        isAvailable = available;
+    }
+
+    public ComponentType getComponentType() {
+        return componentType;
+    }
+
+    public void setComponentType(ComponentType componentType) {
+        if (!componentType.getDeclaringClass().isInstance(ComponentType.class)){
```

Comment:
Podla mna zbytocna podmianka, a ked uz dal by som iba if (EnumSet.allOf(ComponentType.class).contains(componentType) ) {} ale aj to mi prode zbytocne

Reply:
No podľa.mna chceš skontrolovať či to čo chceš nastaviť je správny typ komponentu, možno by som dokonca dal aby kontroloval aj že či už je nastavený a potom nepovolil nastaviť niečo iné

Reply:
Robo ma pravdu, takto je to useless. A Riso ak to chces tak ako pises, tak to prepis nech kontroluje null